cmd/kots/cli/template.go

```diff
@@ -0,0 +1,323 @@
+			if err != nil {
+				return errors.Wrap(err, "failed to parse --license-file")
+			}
+
+			config, err := pull.ParseConfigValuesFromFile(configFile)
+			if err != nil {
+				return errors.Wrap(err, "failed to parse --config-values")
+			}
+
+			configCtx, err := createConfigContext(config)
+			if err != nil {
+				return errors.Wrap(err, "failed to create config context")
+			}
+
+			// TODO: support other contexts
+			builderOptions := template.BuilderOptions{
+				ExistingValues: configCtx,
+				License:        license,
+				DecryptValues:  true,
+			}
+
+			builder, _, err := template.NewBuilder(builderOptions)
+			if err != nil {
+				return errors.Wrap(err, "failed to create template builder")
+			}
+
+			log := logger.NewCLILogger(cmd.OutOrStdout())
+			log.Initialize()
+
+			// when no args are provided
+			if len(args) == 0 && !interactive {
+				// render --data if provided
+				if data != "" {
+					rendered, err := builder.String(data)
+					if err != nil {
+						return errors.Wrap(err, "failed to render raw template")
+					}
+					log.Info(rendered)
+					return nil
+				}
+
+				// render all mode, similar to helm template
+				// we will utilize pull command to fetch and render manifests from upstream
+				if localPath != "" {
+					log.Info("Rendering templates from local path...")
+				} else {
+					log.Info("Pulling app from upstream and rendering templates...")
+				}
+				err := pullAndRender(license.Spec.AppSlug, licenseFile, configFile, localPath)
+
+				if err != nil {
+					return errors.Wrap(err, "failed to render all templates")
+				}
+
+				return nil
+			}
+
+			// interactive mode
+			if interactive {
+				err := runInteractive(&builder, log)
+				if err != nil {
+					return errors.Wrap(err, "failed to run interactive mode")
+				}
+				return nil
+			}
+
+			// non-interactive mode
+			// first argument is path to template file
+			templateFile := args[0]
+			if _, err := os.Stat(templateFile); os.IsNotExist(err) {
+				return errors.Wrap(err, "file does not exist")
+			}
+
+			templateContent, err := os.ReadFile(templateFile)
+			if err != nil {
+				return errors.Wrap(err, "failed to read template file")
+			}
+
+			rendered, err := builder.String(string(templateContent))
+			if err != nil {
+				return errors.Wrap(err, "failed to render template")
+			}
+
+			log.Info(rendered)
+
+			return nil
+		},
+	}
+
+	cmd.Flags().String("license-file", "", "path to a license file to use when download a replicated app")
+	cmd.Flags().String("config-values", "", "path to a manifest containing config values (must be apiVersion: kots.io/v1beta1, kind: ConfigValues)")
+	cmd.Flags().String("data", "", "raw template data to render")
+	cmd.Flags().Bool("interactive", false, "provides an interactive command-line console for evaluating template values")
+	cmd.Flags().String("local-path", "", "specify a local-path to pull a locally available replicated app (only supported on replicated app types currently)")
```

Comment:
what happens if this flag isn't provided?

Is the app not pulled, or is it pulled to the current path?

Reply:
oh I see this is to not pull but to use the app from the local path
```suggestion
	cmd.Flags().String("local-path", "", "the path to a directory containing replicated app yaml to be rendered.")
```